dig/README.md

```diff
@@ -0,0 +1,78 @@
+# Dependency Injection Graph
+
+This package provides a fairly opinionated way of resolving object dependencies.
+There are two sides of dig: `Inject` and `Resolve`
+
+## Inject
+
+Adding an object, or a constructor of an object to the graph.
+
+There are two ways of providing an object:
+
+1. Injecting a pointer to an existing object
+1. Injecting a "constructor function" that returns one pointer (or interface)
+
+### Injecting an object
+
+Injecting an object means it has no dependencies, and will be used as a
+**shared** singleton instance for all resolutions within the graph
+
+```go
+type Fake struct {
+    Name string
+}
+err := g.Inject(&Fake{Name: "I am an injected thing"})
```

Comment:
I was just a bit confused that `Inject` actually instantiated the thing, when in reality it was just providing it to the graph so that when `Resolve` was called, it had everything it needed.

In the context of a regular app, I like this terminology:

1. A DIG graph is created to manage the application graph
2. Structs and constructors are provided to the graph
3. The graph is resolved, resulting in the application graph being fully instantiated